src/nools/task-emitter.js

```diff
@@ -120,10 +120,17 @@ function emitTasks(taskDefinition, Utils, Task, emit, c, r) {
         readyStart: event.start || 0,
         readyEnd: event.end || 0,
         title: taskDefinition.title,
-        resolved: taskDefinition.resolvedIf(c, r, event, dueDate, scheduledTaskIdx),
         actions: taskDefinition.actions.map(initActions),
       };
 
+      if (typeof taskDefinition.resolvedIf === 'function') {
+        task.resolved = taskDefinition.resolvedIf(c, r, event, dueDate, scheduledTaskIdx);
+      }
+      else {
+        var resolvingForm = taskDefinition.actions.find(function (action) { return action.type === 'report'; }).form;
```

Comment:
> I think you can leave it as `defaultResolvedIf` responsibility of how to extract the form.
So we would have to pass the `taskDefinition` as a parameter to the `defaultResolvedIf`, right?

> Also the `find' function can return undefined
It is actually checked in `validate-declarative-schema.js` that if `resolvedIf` is not given, there should be at least one such report. Might not hurt to check it in case the validate logic changes. Thanks.

Reply:
> So we would have to pass the taskDefinition as a parameter to the defaultResolvedIf, right?

I think you can pass the `taskDefinition.actions` instead... unless you need more from taskDefinition.

> It is checked in validate-declarative-schema.js 

Awesome! Please add a unit test to address the scenario if possible :)

Reply:
As the `actions[n].type` is optional, I am thinking of changing it to:
```js
function getDefaultResolvingForm(taskDefinition) {
  var resolvingAction = taskDefinition && taskDefinition.actions &&
    taskDefinition.actions.find(
      function (action) {
        return action.type === 'report' || typeof action.type === 'undefined';
      }
    );
  return resolvingAction && resolvingAction.form;
}
```
Any suggestions?

Reply:
I think you can also check it as falsy value, it can compact to:
```
taskDefinition.actions.find(action => action.type === 'report' || !action.type);
```
if action.type is missing, does it get a default value or it's 'undefined' ?

Reply:
It is `undefined` at this point but I take your suggestion. Thanks!